File: C#/DataStructures/Models/Stack.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace DataStructures.Models
{
    public class Stack<T>
    {
        public int MAXLEN { get; set; }

        public int top { get; set; }
        public T[] elements { get; set; }

        public Stack()
        {
            top = -1;
            elements = new T[MAXLEN];
            MAXLEN = 5;
        }

        private bool reachedMaxCapacity {
            get {
                return top == MAXLEN - 1;
            }
        }

        public void push(T element) {
            if (reachedMaxCapacity) {
                throw new Exception("Stack reached max capacity!");
            }
            
            top += 1;
            elements[top] = element;
        }

        public T pop() {
            if (top == -1) {
                throw new Exception("Oops... The stack is empty! Please add a few elements :)");
            }
            var foos = new List<int>();
            foos.Add(2);
            foos.RemoveAt(2);

            var poppedElement = elements[top];
            top -= 1;

            return poppedElement;
        }

        public T[] viewElements() {
            return elements;
        }
    }
}

File: Algorithms/src/app/data-structures/stack.ts
export class Stack<T> {

    private MAXLEN: number = 50;
    private elements: Array<T> = [];
    private top: number = -1;

    push(element: T): void {
        if (this.reachedMaxCapacity) {
            throw new Error("Stack Overflow Exception: The stack has reached max capacity!");
        }

        this.top += 1;
        this.elements[this.top] = element;
    }

    pop(): T {
        if (this.stackHasNoElements) {
            throw new Error("Stack Underflow Exception: The stack has no elements!");
        }

        const poppedElement = this.elements[this.top];
        this.elements.splice(this.top, 1);
        this.top -= 1;

        return poppedElement;
    }

    get elementCount(): number {
        return this.elements.length;
    }

    get reachedMaxCapacity(): boolean {
        return this.top === (this.MAXLEN - 1);
    }

    get stackHasNoElements(): boolean {
        return this.top === -1;
    }

    get viewElements(): Array<T> {
        return [...this.elements];
    }

    get maxLength(): number {
        return this.MAXLEN;
    }
    set size(stackSize: number) {
        this.MAXLEN = stackSize;
    }
}
